stan/math/prim/mat/fun/scale_matrix_exp_multiply.hpp

```diff
@@ -51,10 +51,11 @@ template <typename Tt, typename Ta, typename Tb, int Cb>
 inline Eigen::Matrix<stan::return_type_t<Tt, Ta, Tb>, -1, Cb>
 scale_matrix_exp_multiply(const Tt& t, const Eigen::Matrix<Ta, -1, -1>& A,
                           const Eigen::Matrix<Tb, -1, Cb>& B) {
-  check_nonzero_size("scale_matrix_exp_multiply", "input matrix", A);
-  check_nonzero_size("scale_matrix_exp_multiply", "input matrix", B);
   check_multiplicable("scale_matrix_exp_multiply", "A", A, "B", B);
   check_square("scale_matrix_exp_multiply", "input matrix", A);
+  if (A.size() == 0 && B.size() == 0) {
+    return {};
+  }
```

Comment:
This should be checked before `check_multiplicable`, because that throws on size 0 inputs.

Reply:
This passed the tests so I am assuming this also needs to be changed in the tests? If so, lets fix this separately. Is that fine with you?

Reply:
Sure, let's merge into develop and then I can take a look.